plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/helper/FormToBpmHelper.java

```diff
@@ -112,28 +111,31 @@ private static JSONObject buildSections(Map<String, JSONObject> jmap,
         logger.debug("INPUT KEYS:  {}", input.keySet().toString());
 
         for (KieProcessFormField formField : formParams) {
+
+            logger.debug("Processing formField {} of type {}: ",formField, formField.getType());
+ 
             String parameter = formField.getName();
             String[] tok = parameter.split(SEPARATOR);
 
             if (tok.length > 1) {
                 String section = tok[0];
                 String field = tok[1];
+                //Boolean checkbox values need to get round tripped as false. Without it data erasure can occur
+                //when a box is un-checked by a user. The input data from the user won't contain a key since the value
+                //isn't sent by the FE
+                if (formField.getType().equals("CheckBox") && !input.containsKey(parameter)) {
```

Comment:
Needs some test coverage here. Even though a small change this is a key output so it needs to be tested